### PR TITLE
JC plugin

### PR DESCRIPTION
**Pull request by kellyjonbrazil**
##### SUMMARY
Add a `jc` filter to `community.general` plugins.

See https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/ansible-devel/5DAEA4uZN0I/zeTjmY3MDAAJ for the genesis of this PR.

`jc` (https://github.com/kellyjonbrazil/jc) is a python library that converts the output of dozens of commands and file types to a dict or a list of dicts. When used as a cli tool it converts the output to JSON. For example:

(CLI)
```
$ arp | jc --arp -p
[
  {
    "address": "gateway",
    "hwtype": "ether",
    "hwaddress": "00:50:56:f7:4a:fc",
    "flags_mask": "C",
    "iface": "ens33"
  },
  {
    "address": "192.168.71.1",
    "hwtype": "ether",
    "hwaddress": "00:50:56:c0:00:08",
    "flags_mask": "C",
    "iface": "ens33"
  },
  {
    "address": "192.168.71.254",
    "hwtype": "ether",
    "hwaddress": "00:50:56:fe:7a:b4",
    "flags_mask": "C",
    "iface": "ens33"
  }
]
```

(Python Libary)
```
>>> import jc.parsers.ls
>>> 
>>> data='''-rwxr-xr-x  1 root  wheel    23648 May  3 22:26 cat
... -rwxr-xr-x  1 root  wheel    30016 May  3 22:26 chmod
... -rwxr-xr-x  1 root  wheel    29024 May  3 22:26 cp
... -rwxr-xr-x  1 root  wheel   375824 May  3 22:26 csh
... -rwxr-xr-x  1 root  wheel    28608 May  3 22:26 date
... -rwxr-xr-x  1 root  wheel    32000 May  3 22:26 dd
... -rwxr-xr-x  1 root  wheel    23392 May  3 22:26 df
... -rwxr-xr-x  1 root  wheel    18128 May  3 22:26 echo'''
>>>
>>> jc.parsers.ls.parse(data)
[{'filename': 'cat', 'flags': '-rwxr-xr-x', 'links': 1, 'owner': 'root', 'group': 'wheel', 'size': 23648,
'date': 'May 3 22:26'}, {'filename': 'chmod', 'flags': '-rwxr-xr-x', 'links': 1, 'owner': 'root',
'group': 'wheel', 'size': 30016, 'date': 'May 3 22:26'}, {'filename': 'cp', 'flags': '-rwxr-xr-x',
'links': 1, 'owner': 'root', 'group': 'wheel', 'size': 29024, 'date': 'May 3 22:26'}, {'filename': 'csh',
'flags': '-rwxr-xr-x', 'links': 1, 'owner': 'root', 'group': 'wheel', 'size': 375824, 'date': 'May 3
22:26'}, {'filename': 'date', 'flags': '-rwxr-xr-x', 'links': 1, 'owner': 'root', 'group': 'wheel',
'size': 28608, 'date': 'May 3 22:26'}, {'filename': 'dd', 'flags': '-rwxr-xr-x', 'links': 1, 'owner':
'root', 'group': 'wheel', 'size': 32000, 'date': 'May 3 22:26'}, {'filename': 'df', 'flags':
'-rwxr-xr-x', 'links': 1, 'owner': 'root', 'group': 'wheel', 'size': 23392, 'date': 'May 3 22:26'},
{'filename': 'echo', 'flags': '-rwxr-xr-x', 'links': 1, 'owner': 'root', 'group': 'wheel', 'size': 18128,
'date': 'May 3 22:26'}]
```
I basically copied the `json_query` filter in `community.general` to create this working proof of concept. This initial version requires the `jc` library to be installed separately (via `pip`, etc.), but `jc` can also be vendored into the collection, if desired.

The filter takes in the raw command output as input and also requires a `parser` argument. For example, when issuing the `uname -a` command on the remote host, you would specify `'uname'` as the `parser` argument for the `jc` filter.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
jc

##### ADDITIONAL INFORMATION
Here is a working example playbook:
```
- name: RUN UBUNTU COMMANDS
  hosts: ubuntu
  tasks:
  - shell: uname -a
    register: result
  - set_fact:
      myvar: "{{ result.stdout | community.general.jc('uname') }}"
  - debug:
      msg: "{{ myvar }}"
```
Here is the output:
```
$ ansible-playbook ubuntu-playbook.yml 

PLAY [RUN UBUNTU COMMANDS] ***********************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************
ok: [192.168.1.239]

TASK [shell] *************************************************************************************************************
changed: [192.168.1.239]

TASK [set_fact] **********************************************************************************************************
ok: [192.168.1.239]

TASK [debug] *************************************************************************************************************
ok: [192.168.1.239] => {
    "msg": {
        "hardware_platform": "x86_64",
        "kernel_name": "Linux",
        "kernel_release": "4.15.0-112-generic",
        "kernel_version": "#113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020",
        "machine": "x86_64",
        "node_name": "kbrazil-ubuntu",
        "operating_system": "GNU/Linux",
        "processor": "x86_64"
    }
}

PLAY RECAP ***************************************************************************************************************
192.168.1.239              : ok=4    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

### CODE REVIEW

**Comment by felixfontein** [comment]
There are two things missing:
1. A changelog fragment (filter plugins are not "documentable"), similar to https://github.com/felixfontein/community.general/blob/8ebcb7887f71a1d1770de13840402d8a9f1d41b6/changelogs/fragments/filter-time.yml (https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to for details on how fragments should look like).
2. You need integration tests. A good starting point are the `json_query` filter tests, with entry point https://github.com/ansible-collections/community.general/blob/main/tests/integration/targets/filter_json_query/runme.sh You can run these either directly in CI, or locally with `ansible-test integration --docker -v filter_jc` (run from the repository root; you should put the tests into `tests/integration/targets/filter_jc`).

**Comment by kellyjonbrazil** [comment]
@felixfontein Tests are passing locally, but failing on CI due to python3.x requirement for `jc`. I tried using `pip3` instead of `pip` to install `jc`, but it shows the following:
```
01:36 + pip3 install jc
01:36 ./runme.sh: line 10: pip3: command not found
```
Not sure of the way forward here.

**Comment by felixfontein** [comment]
Always use `pip`, not `pip3`.

It looks like before, the tests that failed where the Python 2 tests. I guess jc does not support Python 2? In that case, you should also add `skip/python2.7` to the `aliases` file for the test.

File: changelogs/fragments/750-jc-new-filter.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- jc - new filter to convert the output of many shell commands and file-types to JSON. Uses the jc library at https://github.com/kellyjonbrazil/jc. For example, filtering the STDOUT output of ``uname -a`` via ``{{ result.stdout | community.general.jc('uname') }}``
```
Comment (felixfontein) [inline changelogs/fragments/750-jc-new-filter.yaml:2]:
```suggestion
- jc - new filter to convert the output of many shell commands and file-types to JSON. Uses the jc library at https://github.com/kellyjonbrazil/jc. For example, filtering the STDOUT output of ``uname -a`` via ``{{ result.stdout | community.general.jc('uname') }}``. Requires Python 3.5+ (https://github.com/ansible-collections/community.general/pull/750).
```

Comment (felixfontein) [reply changelogs/fragments/750-jc-new-filter.yaml:2]: (3.5 is the minimal Python 3 version supported by Ansible)

Comment (kellyjonbrazil) [reply changelogs/fragments/750-jc-new-filter.yaml:2]: There are a bunch of f-strings littered about `jc` so I believe only python 3.6+ is supported. I'll update to 3.6.

**Comment by kellyjonbrazil** [comment]
Ok - I think we are all set now.

**Comment by Andersson007** [comment]
@kellyjonbrazil thanks for adding the new feature!
@felixfontein thanks for reviewing!
merged

**Pull request merged**